### DISCUSSION TITLE
ListBox.ItemTemplate - Bug or Feature?

### DISCUSSION THREAD

**Discussion by K2kmoments** [AvaloniaUI/Avalonia#15763] [Q&A]
<img width="1692" alt="Screenshot 2024-05-19 at 09 53 09" src="https://github.com/AvaloniaUI/Avalonia/assets/29265695/c969b922-e276-49e2-9dc3-e95e86b6245b">

Ladies and Gentleman, I am having a question Regarding the ListView / ListBoxTemplate:

In my "PacketSender Clone Training Application" I want to make a ItemTemplate to fill it with a list. Every time when I delete the **">"** (marked in the screenshot,) the list view in the preview and built application disappears.

- To be honest, it was only a accident that I pressed the ">" symbol, before that I never managed to make the whole listBox.ItemTemplate visible.
- uhhmm.. why?

greets

**Comment by stevemonaco** [comment]
This is probably by-design. I imagine the single child item (<) is getting wrapped into a `ListBoxItem` and being added to children. Which means it then has one item and displays one copy of the `DataTemplate`. Empty collection controls won't show anything because there are no items to render.

This is why I don't use the Previewer for app views: you need to create design-time data to keep the Previewer remotely realistic. Just run your app under normal circumstances and/or use HotAvalonia.

**Comment by timunie** [comment]
There is a Design.DataContext which you can use to feed the previewer with some test data if needed.
